ElectronClient/gui/KeymapConfig/KeymapConfigScreen.tsx

```diff
@@ -0,0 +1,144 @@
+import * as React from 'react';
+import { useState } from 'react';
+
+import KeymapService, { KeymapItem } from '../../lib/services/KeymapService';
+import { ShortcutRecorder } from './ShortcutRecorder';
+import getLabel from './utils/getLabel';
+import useKeymap from './utils/useKeymap';
+import useEditing from './utils/useEditing';
+import styles_ from './styles';
+
+const { _ } = require('lib/locale.js');
+const keymapService = KeymapService.instance();
+
+export interface KeymapConfigScreenProps {
+	themeId: number
+}
+
+export const KeymapConfigScreen = ({ themeId }: KeymapConfigScreenProps) => {
+	const styles = styles_(themeId);
+
+	const [filter, setFilter] = useState('');
+	const [keymap, setAccelerator, error] = useKeymap();
+	const [editing, enableEditing, disableEditing] = useEditing();
+
+	const handleSave = (event: { commandName: string, accelerator: string }) => {
+		const { commandName, accelerator } = event;
+		setAccelerator(commandName, accelerator);
+		disableEditing(commandName);
+	};
+
+	const hanldeReset = (event: { commandName: string }) => {
+		const { commandName } = event;
+		setAccelerator(commandName, keymapService.getDefaultAccelerator(commandName));
+		disableEditing(commandName);
+	};
+
+	const handleCancel = (event: { commandName: string }) => {
+		const { commandName } = event;
+		disableEditing(commandName);
+	};
+
+	const renderAccelerator = (accelerator: string) => {
+		return (
+			<div>
+				{accelerator.split('+').map(part => <kbd style={styles.kbd} key={part}>{part}</kbd>).reduce(
+					(accumulator, part) => (accumulator.length ? [...accumulator, ' + ', part] : [part]),
+					[]
+				)}
+			</div>
+		);
+	};
+
+	const renderStatus = (command: string) => {
+		if (!error) return null;
+		const { invalidAccelerator, invalidKeymapItem, duplicateAccelerators, duplicateKeymapItems } = error.details;
+
+		if ((invalidAccelerator && invalidKeymapItem.command === command) ||
+			(duplicateAccelerators && (duplicateKeymapItems[0].command === command || duplicateKeymapItems[1].command === command))) {
+			return '❌';
```

Comment:
This will render differently and sometimes strangely depending on the operating system so please use an icon (Font Awesome is always loaded).